api/handlers/configure.go

```diff
@@ -0,0 +1,304 @@
+package handlers
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/go-openapi/swag"
+	networkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	v1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	versionedclient "istio.io/client-go/pkg/clientset/versioned"
+
+	"github.com/go-openapi/runtime/middleware"
+	keptnutils "github.com/keptn/go-utils/pkg/lib"
+	"github.com/keptn/keptn/api/models"
+	"github.com/keptn/keptn/api/restapi/operations/configure"
+	k8sutils "github.com/keptn/kubernetes-utils/pkg"
+	networking "k8s.io/api/networking/v1beta1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Ingress is an enum type for the ingress
+type Ingress int
+
+const (
+	istio Ingress = iota
+	nginx
+)
+
+const keptnGateway = "public-gateway.istio-system"
+
+func (i Ingress) String() string {
+	return [...]string{"istio", "nginx"}[i]
+}
+
+const useInClusterConfig = true
+
+func getIngressType() (Ingress, error) {
+
+	clientset, err := k8sutils.GetKubeAPI(useInClusterConfig)
+	if err != nil {
+		return istio, err
+	}
+
+	nsList, err := clientset.Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return istio, err
+	}
+
+	for _, ns := range nsList.Items {
+		if ns.Name == "istio-system" {
+			return istio, nil
+		} else if ns.Name == "ingress-nginx" {
+			return nginx, nil
+		}
+	}
+	return istio, errors.New("Cannot obtain type of ingress.")
```

Comment:
Are you returning `istio` on purpose here? Or should this be nil?

Reply:
The type `Ingress` cannot be nil.